neurom/features/morphology.py

```diff
@@ -480,14 +483,25 @@ def sholl_crossings(morph, neurite_type=NeuriteType.all, center=None, radii=None
     """
     def _count_crossings(neurite, radius):
         """Used to count_crossings of segments in neurite with radius."""
-        r2 = radius ** 2
         count = 0
-        for start, end in iter_segments(neurite):
-            start_dist2, end_dist2 = (morphmath.point_dist2(center, start),
-                                      morphmath.point_dist2(center, end))
 
-            count += int(start_dist2 <= r2 <= end_dist2 or
-                         end_dist2 <= r2 <= start_dist2)
+        if distance_type == 'euclidean':
+            r2 = radius ** 2
+            for start, end in iter_segments(neurite):
+                start_dist2, end_dist2 = (morphmath.point_dist2(center, start),
+                                          morphmath.point_dist2(center, end))
+
+                count += int(start_dist2 <= r2 <= end_dist2 or
+                             end_dist2 <= r2 <= start_dist2)
+
+        if distance_type == 'path':
+            for section in iter_sections(neurite):
+                base_length = sf.section_path_length(section) - section.length
+                path_dists = base_length + sf.segment_lengths(section, prepend_zero=True)
+
+                forward = np.logical_and(path_dists[:-1] <= radius, path_dists[1:] >= radius)
+                backward = np.logical_and(path_dists[:-1] >= radius, path_dists[1:] <= radius)
```

Comment:
On second thought, this doesn't make much sense, does it?

Radial distance and path distance are two incompatible metrics. However, here are compared for the Sholl crossings, because you are using the radial distance of the shell against the path distance of the segments.

You may want to refer the the sholl descriptors in the following paper:
https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010229

at the 2.2 section, point 7

> Total Wiring. Take a tree being a connected component of a neuron that contain soma, restricted to a ball of a radius r. This construction measures the total path length of this tree. This is the sum of the path distances of all dendritic segments and it is related to the notion of Arbor Density or Arbor Territory [[43](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010229#pcbi.1010229.ref043)]. Greater total wiring dendritic length means greater invaded space which render the dendrites available to more incoming connections and greater connectivity.

Reply:
I'm not 100% sure I follow you, but now the 'shell' with radius `radius` is in path length, so it's not a spherical shell in 3d. I think this does what is in the paper you mention, no?

Reply:
`radius` is not reflecting path length because the bins are calculated from the radial distance in the sholl frequency function:
```python
bins = np.arange(min_soma_edge, min_soma_edge + max(max_radius_per_neurite), step_size)
```
In the paper, it states indeed that you can have path distance descriptors with an independent var on the path length span L(N), and then they proceed to list the most common descriptors along with their physical interpretation. All of them use radial distance as the independent variable, including the total wiring I mentioned above that gives insight into arborization density.

Therefore, I wonder what the physical meaning of this descriptor is.

Reply:
ah yes, good catch, I forgot to update the frequency bins, now it's consistent